pkg/providers/instance/instance.go

```diff
@@ -78,15 +83,19 @@ func NewProvider(ctx context.Context, region string, ec2api ec2iface.EC2API, una
 		subnetProvider:         subnetProvider,
 		launchTemplateProvider: launchTemplateProvider,
 		ec2Batcher:             batcher.EC2(ctx, ec2api),
+		recorder:               recorder,
 	}
 }
 
 func (p *Provider) Create(ctx context.Context, nodeClass *v1beta1.EC2NodeClass, nodeClaim *corev1beta1.NodeClaim, instanceTypes []*cloudprovider.InstanceType) (*Instance, error) {
+	filteredITsRequirement := len(instanceTypes)
```

Comment:
I feel like we still want to understand the instance types that were filtered out since they didn't match the requirements, or the ones that weren't large enough with the `resource.Fits` calculation, or the ones that didn't have any compatible offerings that were available when we went to go resolve the instance types. Are there ways that we could break up the `resolveInstanceTypes` function in a way that would also allow us to understand if we filtered anything out there. Otherwise, we are flying a little blind around why we might have filtered out an instance type when we went to first resolve them

Reply:
This number is also wrong, right? This isn't the number of instance types that were filtered out after requirements, this is the number of instance types that work after the requirements has filtered them out?

Reply:
Musing: Is there one function that we could build to combine all of the filtering logic into one place so that capturing all of this information and creating an event is much easier and is self-contained to just that function?